core/blockchain.go

```diff
@@ -274,7 +274,11 @@ func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *par
 	if err != nil {
 		return nil, err
 	}
-	bc.genesisBlock = bc.GetBlockByNumber(0)
+	if chainConfig.IsArbitrum() {
+		bc.genesisBlock = bc.GetBlockByNumber(chainConfig.ArbitrumChainParams.GenesisBlockNum)
+	} else {
+		bc.genesisBlock = bc.GetBlockByNumber(0)
```

Comment:
I'm a bit concerned about this change as it impacts everyone querying for the genesis block (which they might assume is block 0), but my preliminary search of the codebase didn't turn up anything dangerous

Reply:
The only place I saw so far that access the genesis block (via Genesis() function), not during a test is forkid, accoring to https://eips.ethereum.org/EIPS/eip-2124 - I suppose there could be arguments both ways, and either block0 or nitro genesis will ultimately be fine.
In tests (e.g. graphql) as well as here I saw genesis used as the first block from which others are built - in which case we definitely want the nitro genesis.